token/core/zkatdlog/nogh/sender.go

```diff
@@ -108,32 +123,42 @@ func (s *Service) Transfer(txID string, wallet driver.OwnerWallet, ids []*token3
 	return transfer, metadata, nil
 }
 
+// VerifyTransfer checks the outputs in the TransferAction against the passed tokenInfos
 func (s *Service) VerifyTransfer(action driver.TransferAction, tokenInfos [][]byte) error {
 	tr, ok := action.(*transfer.TransferAction)
 	if !ok {
-		return errors.Errorf("expected *zkatdlog.Transfer")
+		return errors.New("expected *zkatdlog.Transfer")
```

Comment:
shall we check if the action is nil as well before the cast?

Reply:
OK